### ISSUE TITLE
Support choosing Vulkan device

### ISSUE THREAD

**Issue by mvainola** [google/amber#619]
Currently Amber picks the first VkPhysicalDevice that meets its feature and extension requirements. Setting `VK_ICD_FILENAMES` is sufficient in most cases but not when trying to choose between two devices using the same ICD.

My idea was to add two new command line options:

* List physical Vulkan devices with IDs and exit
* Specify Vulkan device ID to run with

**Comment by dj2** [comment]
I'd skip the first one, if you can get the vulkan devices from the vulkan info command, yea? The second seems like it would be useful to add.

**Comment by mvainola** [comment]
I'm fine with using vulkaninfo to get the IDs and just adding an option similar to --deqp-vk-device-id in the CTS.

**Comment by dj2** [comment]
Let's go with something like `-d` for device. That way, hopefully, it can be used for Dawn as well. @sarahM0 is there a way in Dawn to specify the device to use?

**Comment by sarahM0** [comment]
@Kangz

**Comment by Kangz** [comment]
Yep that's something we do in the `dawn_end2end_tests` already. See mVendorIDFilter and friend [here](https://cs.chromium.org/chromium/src/third_party/dawn/src/tests/DawnTest.cpp?sq=package:chromium&g=0&l=120) that use `dawn_native::Adapter::GetPCIInfo` to get information to do the filtering.

**Comment by dj2** [comment]
Hm, didn't mean to close, #698 does the Vulkan side, still need Dawn support
